src/micro_http/src/common/headers.rs

```diff
@@ -134,7 +134,7 @@ impl Headers {
                 }
                 if let Ok(head) = Header::try_from(entry[0].as_bytes()) {
                     match head {
-                        Header::ContentLength => match entry[1].trim().parse::<i32>() {
+                        Header::ContentLength => match entry[1].trim().parse::<u32>() {
```

Comment:
nit: as a double-check, I would add the following:

```rust
        // Invalid content length.
        assert_eq!(
            header.parse_header_line(b"Content-Length: -1"),
            Err(RequestError::InvalidHeader)
        );
```
in `test_parse_header_line()`, even though this is tested in `connection.rs` too.